framework/PostProcessors/ETstructure.py

```diff
@@ -0,0 +1,482 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on April 30, 2018
+
+@author: mandd
+"""
+
+from __future__ import division, print_function , unicode_literals, absolute_import
+import warnings
+warnings.simplefilter('default', DeprecationWarning)
+
+#Internal Modules---------------------------------------------------------------
+import MessageHandler
+from utils import utils
+from utils import xmlUtils as xmlU
+#Internal Modules End-----------------------------------------------------------
+
+#External Modules---------------------------------------------------------------
+import numpy as np
+import xml.etree.ElementTree as ET
+import copy
+import itertools
+from collections import OrderedDict
+#External Modules End-----------------------------------------------------------
+
+class ETstructure():
+  """
+    This is the base class of the ET structure which actually handles ET structures which is used by the ETimporter and the ETmodel
+  """
+  def __init__(self, expand, inputs):
+    """
+      This method executes the post-processor action.
+      @ In,  inputs, list, list of file objects
+      @ Out, None
+    """
+    self.expand = expand
+    ### Check for link to other ET
+    links = []
+    sizes=(len(inputs),len(inputs))
+    connectivityMatrix = np.zeros(sizes)
+    listETs=[]
+    listRoots=[]
+
+    for file in inputs:
+      eventTree = ET.parse(file.getPath() + file.getFilename())
+      listETs.append(eventTree.getroot().get('name'))
+      listRoots.append(eventTree.getroot())
+    links = self.createLinkList(listRoots)
+
+    if len(inputs)>0:
+      rootETID = self.checkETstructure(links,listETs,connectivityMatrix)
+
+    if len(links)>=1 and len(inputs)>1:
+      finalAssembledTree = self.analyzeMultipleET(inputs,links,listRoots,listETs,rootETID)
+      self.pointSet = self.analyzeSingleET(finalAssembledTree)
+
+    if len(links)==0 and len(inputs)>1:
+      raise IOError('Multiple ET files have provided but they are not linked')
```

Comment:
why you do not use the message handler?

Reply:
The class ```ETstructure``` is used in both ETModel and ETImport. ETModel is an external model plugin. Currently, external model plugin are not inherited from the message handler. Thus, we can not use message handler in the ```ETstructure```.